server/controllers/events/handlers/pull_request_review.go

```diff
@@ -2,18 +2,54 @@ package handlers
 
 import (
 	"context"
+	"github.com/runatlantis/atlantis/server/events/models"
 	"github.com/runatlantis/atlantis/server/http"
+	"github.com/runatlantis/atlantis/server/logging"
 	"github.com/runatlantis/atlantis/server/neptune/gateway/event"
+	"time"
 )
 
-type PullRequestReviewEvent struct {
+type PRReviewCommandRunner interface {
+	RunPRReviewCommand(ctx context.Context, repo models.Repo, pull models.PullRequest, user models.User, timestamp time.Time, installationToken int64)
 }
 
-func (p PullRequestReviewEvent) Handle(ctx context.Context, e event.PullRequestReview, _ *http.BufferedRequest) error {
-	//TODO implement me to run a modified approve command runner
+type PullRequestReviewEventHandler struct {
+	PRReviewCommandRunner PRReviewCommandRunner
+}
+
+func (p PullRequestReviewEventHandler) Handle(ctx context.Context, event event.PullRequestReview, _ *http.BufferedRequest) error {
+	p.PRReviewCommandRunner.RunPRReviewCommand(
+		ctx,
+		event.Repo,
+		event.Pull,
+		event.User,
+		event.Timestamp,
+		event.InstallationToken,
+	)
 	return nil
 }
 
-func NewPullRequestReviewEvent() *PullRequestReviewEvent {
-	return &PullRequestReviewEvent{}
+type AsyncPullRequestReviewEvent struct {
+	handler *PullRequestReviewEventHandler
+	logger  logging.Logger
+}
+
+func NewPullRequestReviewEvent(prReviewCommandRunner PRReviewCommandRunner, logger logging.Logger) *AsyncPullRequestReviewEvent {
+	return &AsyncPullRequestReviewEvent{
+		handler: &PullRequestReviewEventHandler{
+			PRReviewCommandRunner: prReviewCommandRunner,
+		},
+		logger: logger,
+	}
+}
+
+func (a AsyncPullRequestReviewEvent) Handle(_ context.Context, event event.PullRequestReview, req *http.BufferedRequest) error {
+	go func() {
+		// Passing background context to avoid context cancellation since the parent goroutine does not wait for this goroutine to finish execution.
+		err := a.handler.Handle(context.Background(), event, req)
+		if err != nil {
+			a.logger.ErrorContext(context.Background(), err.Error())
+		}
+	}()
```

Comment:
you have a context here why not pass it through?

Reply:
see thread: https://github.com/lyft/atlantis/pull/254#discussion_r882953595